instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb

```diff
@@ -27,8 +27,18 @@ class GraphQLTracer < ::GraphQL::Tracing::PlatformTracing
           def platform_trace(platform_key, key, data)
             return yield if platform_key.nil?
 
-            tracer.in_span(platform_key, attributes: attributes_for(key, data)) do |_span|
-              yield
+            tracer.in_span(platform_key, attributes: attributes_for(key, data)) do |span|
+              yield.tap do |response|
+                if key == 'validate' && !response[:errors]&.empty?
+                  span.status = OpenTelemetry::Trace::Status.error
+                  response[:errors].each do |error|
+                    span.add_event(
+                      'graphql error',
+                      attributes: error.respond_to?(:to_h) ? { 'error' => error.to_h.to_json } : nil
```

Comment:
If you want greater granularity here, you could use `message`, `locations`, or `path`, which are described in the spec: https://spec.graphql.org/draft/#sec-Errors.Error-result-format

Reply:
https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/static_validation/error.rb

I think its ok to rely on `https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/static_validation/error.rb` `to_h`. It will include `message, locations, path`